### PR TITLE
Reset PTO and loss detection timer when discarding keys

### LINKED ISSUE #3272: pto_count should be reset when dropping a packet number space
`pto_count` should be reset to 0 when a packet number space is dropped. I'd argue that this is the consistent thing to do, because when we drop a packet number space, we implicitely acknowledge all outstanding packets in that space (and remove them from `bytes_in_flight`).

### PR DESCRIPTION
**Pull request by ianswett**
Fixes #3272

### CODE REVIEW

**Comment by ianswett** [comment]
Pseudocode added, PTAL

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -1432,6 +1434,26 @@ Invoked from DetectLostPackets when packets are deemed lost.
...
+## Upon dropping Initial or Handshake keys
+
+When Initial or Handshake keys are discarded, packets from the space
+are discarded and loss detection state is updated.
+
+Pseudocode for OnPacketNumberSpaceDiscarded follows:
+
+~~~
+OnPacketNumberSpaceDiscarded(pn_space):
+  assert(pn_space != 1RTT)
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md]: Don't you use ApplicationData?

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -1432,6 +1434,26 @@ Invoked from DetectLostPackets when packets are deemed lost.
...
+  // Remove any unacknowledged packets from flight.
+  foreach packet in sent_packets[pn_space]:
+    if packet.in_flight
+      bytes_in_flight -= size
+      packet.in_flight = false
```
[the remark this answers is not in the document]
Comment (ianswett) [reply draft-ietf-quic-recovery.md]: Good point, I changed it to clear().

File: draft-ietf-quic-recovery.md (hunk 1 above)
Comment (ianswett) [reply draft-ietf-quic-recovery.md]: Of course, fixed.

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -531,6 +531,9 @@ bytes.
 
 Initial packets and Handshake packets could be never acknowledged, but they are
 removed from bytes in flight when the Initial and Handshake keys are discarded.
+When Initial or Handshake keys are discarded, the PTO and loss detection timers
+are reset, because discarding keys indicates forward progress and the loss
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]:
```suggestion
MUST be reset, because discarding keys indicates forward progress and the loss
```

**Comment by ianswett** [comment]
I'm fine with requiring it, given it's already a design issue.

**Pull request merged**